go/activity.go

```diff
@@ -0,0 +1,37 @@
+package app
+
+import (
+	"context"
+	"fmt"
+	"time"
+)
+
+func BookHotel(ctx context.Context, bookingInfo BookingInfo, start time.Time, end time.Time, idempotencyKey string) error {
```

Comment:
Recommendation for activities with many params is to put them all into a struct, say 

```suggestion
// (put this in shared.go)
type BookHotelOptions struct {
  BookingInfo BookingInfo
  Start time.Time
  End time.Time
  IdempotencyKey string
}

func BookHotel(ctx context.Context, booking BookHotelOptions) error {
```